include/fmt/ranges.h

```diff
@@ -340,29 +340,36 @@ template <typename Context, typename Element> struct range_mapper {
 };
 
 template <typename Char, typename Element>
-using range_formatter_type =
-    conditional_t<is_formattable<Element, Char>::value,
-                  formatter<remove_cvref_t<decltype(
-                                range_mapper<buffer_context<Char>, Element>{}
-                                    .map(std::declval<Element>()))>,
-                            Char>,
-                  fallback_formatter<Element, Char>>;
+using range_formatter_type = conditional_t<
+    is_formattable<Element, Char>::value,
+    formatter<remove_cvref_t<decltype(range_mapper<buffer_context<Char>>{}.map(
+                  std::declval<Element>()))>,
+              Char>,
+    fallback_formatter<Element, Char>>;
+
+template <typename R>
+using maybe_const_range =
+    conditional_t<has_const_begin_end<R>::value, const R, R>;
 }  // namespace detail
 
 template <typename R, typename Char>
 struct formatter<
     R, Char,
-    enable_if_t<fmt::is_range<R, Char>::value
+    enable_if_t<
+        fmt::is_range<R, Char>::value
 // Workaround a bug in MSVC 2019 and earlier.
 #if !FMT_MSC_VER
-                && (is_formattable<detail::uncvref_type<R>, Char>::value ||
-                    detail::has_fallback_formatter<detail::uncvref_type<R>,
-                                                   Char>::value)
+        &&
+        (is_formattable<detail::uncvref_type<detail::maybe_const_range<R>>,
+                        Char>::value ||
+         detail::has_fallback_formatter<
+             detail::uncvref_type<detail::maybe_const_range<R>>, Char>::value)
 #endif
-                >> {
+        >> {
 
+  using const_range = detail::maybe_const_range<R>;
```

Comment:
`const_range` seems a bit misleading considering that it can be non-const. Maybe rename to `range_type`?

Reply:
Agreed. Renamed.